c++/include/orc/Reader.hh

```diff
@@ -39,6 +39,8 @@ namespace orc {
   // classes that hold data members so we can maintain binary compatibility
   struct ReaderOptionsPrivate;
   struct RowReaderOptionsPrivate;
+  struct CacheOptions;
```

Comment:
`CacheOptions` is defined in src/Cache.hh, which is not installed so that downstream users do not have access to it. I'd suggest to choose one of these options:
1. Move the definition of CacheOptions to this file.
2. Directly adding two parameters to ReaderOptions and totally remove CacheOptions.

Personally I'd prefer the 2nd option.

Reply:
I think `CacheOptions` being exposed to user is not a big deal.

Reply:
OK. A side note is that perhaps this is more like a IO coalescing options since the cache instance is not visible to end user.